tools/ci/tc/github_checks_output.py

```diff
@@ -8,22 +16,25 @@ class GitHubChecksOutputter(object):
     See https://docs.taskcluster.net/docs/reference/integrations/github/checks#custom-text-output-in-checks
     """
     def __init__(self, path):
+        # type: (Text) -> None
         self.path = path
 
     def output(self, line):
+        # type: (Any) -> None
```

Comment:
Maybe AnyStr?

Reply:
In fact, if we want to future proof this, we'd better take `Text` or coerce line into Text, open the file in `at` mode, and write `u'\n'`.

Reply:
Why not `Text`? Do we actually pass bytes into the logging ever?

Reply:
As it stands right now, we are using a bunch of native strings.

Reply:
I would love to, but we can't. With `Text`, it fails mypy on py2 mode:

`tools/ci/tc/github_checks_output.py:23: error: Argument 1 to "write" of "IO" has incompatible type "unicode"; expected "str"`

Why is this, write() should take either string type? reveal_type(f) gives:

`tools/ci/tc/github_checks_output.py:23: note: Revealed type is 'typing.BinaryIO'`

What?! It's `TextIO`, not `BinaryIO`. That is exactly what https://github.com/python/typeshed/pull/4146 was meant to fix. We have a mode that is in their list of `TextIO` modes ('a', so is 'at' and that also doesn't work...). Poked at this for a while, couldn't fix it.

Ok, so give up on getting the right type for `f`. What can we do about this line instead.

* We can't use `AnyStr`, because that's still not a match to what `BinaryIO` wants.
* We can't cast `BinaryIO` because it claims incompatible types.
* And we can't type: ignore it, because py3 lint complains about an unused 'type: ignore'.

So instead, `Any`.

Reply:
Well, the said typeshed change wasn't included in the latest mypy release (0.782): https://github.com/python/mypy/commits/release-0.780

Maybe wait for 0.790 to release? https://github.com/python/mypy/issues/9290 seems to suggest it's coming soon.

Reply:
That issue sounds like it's going to be delayed and that the release could eventually break things :/ Let's go with `Any` and a comment on why for now.